File: consumer-verticle/ruby-blur/Worker.rb
require "vertx"
include Vertx

id = Vertx::EventBus.register_handler('image.transform.blur') do |message|
    finalPath = message.body['destinationFilePath']
    command="convert #{message.body['originalFilePath']} -blur 8x8 #{finalPath}"
    system "#{command}"

    Vertx::EventBus.send("image.processing.completed",{
     "name"=>message.body['name'],
     "updatedName"=>message.body['updatedName']
    })

    puts "#{finalPath} :: Finished"
    puts "------------------------------------------"
end

File: consumer-verticle/python-pixelate/Worker.py
import vertx
import os

from core.event_bus import EventBus


def myHandler(message):
    finalPath = message.body['destinationFilePath']
    command= ("convert -scale 10% -scale 1000% " + message.body['originalFilePath'] + " " +  finalPath)
    os.system(command)

    EventBus.send("image.processing.completed",{'updatedName':message.body['updatedName'],'name':message.body['name']})

    print finalPath +" ::Finished"
    print "------------------------------------------"

EventBus.register_handler('image.transform.pixelate', handler=myHandler)